test/main.test.js

```diff
@@ -0,0 +1,286 @@
+const test = require('tap').test;
+const path = require('path');
+const fs = require('fs');
+const request = require('request');
+const httpServer = require('../lib/http-server');
+const promisify = require('util').promisify;
+
+const requestAsync = promisify(request);
+const fsReadFile = promisify(fs.readFile);
```

Comment:
What do we gain by promisify-ing rather than using the promise-based `fs` functions?

Reply:
Node 8 compatibility (currently Travis is set to run Node 8,10,12,13,14)

Reply:
Alright, I'm alright with that, I forgot `fs.promises` is that new. Node 8 is no longer supported but we might as well test it.